e2e/features/steps/register/common.steps.ts

```diff
@@ -0,0 +1,39 @@
+import { Given, When } from "@cucumber/cucumber";
+
+import { page } from "../../../utils/index";
+import { baseUrl } from "../../../utils/consts";
+
+When("the user enters a name: {string}", async function (username: string) {
+  await page.getByText(/^Név$/).first().fill(username);
+});
+
+When("the user enters a password: {string}", async function (password: string) {
+  await page.getByTitle("Jelszó").first().fill(password);
+});
+
+When(
+  "the user enters an email address: {string}",
+  async function (email: string) {
+    await page
+      .getByText(/^Email$/)
+      .first()
+      .fill(email);
+  },
+);
+
+When(
+  "the user clicks the {string} button",
+  async function (buttonLabel: string) {
+    const submitButtonLocator = page
+      .getByRole("button", { name: buttonLabel })
+      .first();
```

Comment:
Im getting the followin error:
```
2) Scenario: Cancel registration # features/registration.feature:22
   ✔ Given the user is on the registration page # features/steps/register/common.steps.ts:34
   ✔ When the user enters a name: "validName" # features/steps/register/common.steps.ts:6
   ✔ And the user enters an email address: "valid@email.com" # features/steps/register/common.steps.ts:14
   ✔ And the user enters a password: "ValidPassword123" # features/steps/register/common.steps.ts:10
   ✖ And the user clicks the "JELENTKEZZEN BE!" button # features/steps/register/common.steps.ts:24
       Error: function timed out, ensure the promise resolves within 5000 milliseconds
           at Timeout.<anonymous> (/workspaces/sk0/e2e/node_modules/@cucumber/cucumber/src/time.ts:52:14)
           at listOnTimeout (node:internal/timers:573:17)
           at processTimers (node:internal/timers:514:7)
   - Then the page should be naviagted to the login page # features/steps/register/cancel.steps.ts:5
 ```
I cant resolve this. No matter what selector or locator I use with any kind of parameters (even tried giving data-testid for both buttons), it wont work for both "Regisztráció" and "Jelentkezzen be" buttons.

Reply:
I made the universal [Button click](https://github.com/cimem-hu/sk0/pull/93/files) function in the common.steps.ts

Reply:
@Thomas851204 you are correct, let me create a commit where I am rearranging the steps, it is a bit counter intuitive now, needs some love